api/coreservice.go

```diff
@@ -243,13 +246,21 @@ func WithAPIStats(stats *nodestats.APILocalStats) Option {
 	}
 }
 
+// WithArchiveSupport is the option to enable archive support
+func WithArchiveSupport(enabled bool) Option {
+	return func(svr *coreService) {
+		svr.archiveSupported = enabled
+	}
+}
+
 type intrinsicGasCalculator interface {
 	IntrinsicGas() (uint64, error)
 }
 
 var (
 	// ErrNotFound indicates the record isn't found
-	ErrNotFound = errors.New("not found")
+	ErrNotFound            = errors.New("not found")
+	ErrArchiveNotSupported = errors.New("archive-mode not supported")
```

Comment:
comment

Reply:
removed L261 comment, the name clears explains what the error is